website/docs/src/content/component-reference/mcp.mdx

```diff
@@ -0,0 +1,129 @@
+| :---------------------------------------------------- | :--------------------------------------------------------------------- |
+| [`createMCPServerContext`](#createmcpservercontext)   | Creates a context provider and hook for accessing MCP server resources |
+| [`MCPTool`](#mcptool)                                 | Represents a tool available in the MCP server                          |
+| [`MCPResource`](#mcpresource)                         | Represents a resource available in the MCP server                      |
+| [`MCPResourceTemplate`](#mcpresourcetemplate)         | Represents a resource template in the MCP server                       |
+| [`MCPPrompt`](#mcpprompt)                             | Represents a prompt available in the MCP server                        |
+
+## Reference
+
+### `createMCPServerContext()`
+
+The `createMCPServerContext` function creates a context provider and hook for accessing MCP server resources. It returns an object containing a Provider component and a useContext hook.
+If a server command is provided, it will be used to start the MCP server, and close the connection when the component is unmounted. Otherwise, the MCP client will be used to connect to an existing server.
+
+```tsx
+import { createMCPServerContext } from "@gensx/mcp";
+
+const { Provider, useContext } = createMCPServerContext({
+  serverCommand: "your-server-command",
+  serverArgs: ["--arg1", "--arg2"],
+  // Or provide a client directly
+  client: yourMCPClient,
+});
+
+// Use the Provider to wrap your application
+<Provider>
+  <YourApp />
+</Provider>;
+
+// Use the context hook in your components
+const MyComponent = () => {
+  const { tools, resources, resourceTemplates, prompts } = useContext();
+  // Use the MCP server context...
+};
+```
+
+#### Parameters
+
+The `createMCPServerContext` function accepts a server definition object with the following properties:
+
+- Either:
+  - `serverCommand`: The command to start the MCP server
+  - `serverArgs`: Array of arguments for the server command
+- Or:
+  - `client`: A pre-configured MCP client instance
+
+#### Return Value
+
+Returns an object containing:
+
+- `Provider`: A React component that provides the MCP server context
+- `useContext`: A hook that returns the current MCP server context
+
+### Types
+
+#### MCPServerContext
+
+The context object returned by `useContext` contains:
+
+```tsx
+interface MCPServerContext {
+  tools: MCPTool[]; // Available tools in the server
+  resources: MCPResource[]; // Available resources
+  resourceTemplates: MCPResourceTemplate[]; // Available resource templates
+  prompts: MCPPrompt[]; // Available prompts
+}
+```
+
+#### MCPTool
+
+Represents a tool available in the MCP server.
+
+#### MCPResource
+
+Represents a resource available in the MCP server.
+
+#### MCPResourceTemplate
+
+Represents a resource template in the MCP server.
+
+#### MCPPrompt
+
+Represents a prompt available in the MCP server.
+
```

Comment:
are these supposed to be just a short sentence each?

Reply:
a minimal code example might be helpful

Reply:
Updated to provide a bit more detail.